src/main/java/hudson/plugins/analysis/core/AnnotationsClassifier.java

```diff
@@ -51,4 +53,9 @@ public ParserResult invoke(final File workspace, final VirtualChannel channel) t
         }
         return result;
     }
+
+    @Override
+    public void checkRoles(RoleChecker checker) throws SecurityException {
+        // TODO required in 1.580 (do we need to implement something here?)
+    }
```

Comment:
You must extend `MasterToSlaveFileCallable` instead. Never implement `checkRoles` yourself.